src/appleseed/renderer/kernel/lighting/pathvertex.h

```diff
@@ -72,6 +73,17 @@ class PathVertex
     const EDF*                  m_edf;
     const BSDF*                 m_bsdf;
     const void*                 m_bsdf_data;
+
+    // BSSRDF data
+    const BSSRDF*               m_bssrdf;
+    const void*                 m_bssrdf_data;
```

Comment:
The reason why we can have `m_bsdf_data` above is that BSDF inputs are evaluated by the path tracing loop (in `pathtracer.h`) and are thus still available when we invoke the path vertex visitor.